raiden-ts/src/channels/epics.ts

```diff
@@ -250,6 +251,39 @@ export function blockStaleEpic(
   );
 }
 
+/**
+ * Monitors provider for current mean gasPrice, and multiply it by config.gasPriceFactor
+ *
+ * @param action$ - Observable of RaidenActions
+ * @param state$ - Observable of RaidenStates
+ * @param deps - RaidenEpicDeps members
+ * @param deps.provider - Provider instance
+ * @param deps.config$ - Config observable
+ * @returns Observable of blockGasprice actions
+ */
+export function blockGasPriceEpic(
+  {}: Observable<RaidenAction>,
+  state$: Observable<RaidenState>,
+  { provider, config$ }: RaidenEpicDeps,
+): Observable<blockGasprice> {
+  return state$.pipe(
+    pluckDistinct('blockNumber'),
+    withLatestFrom(config$),
+    // switchMap will "reset" timer every block, restarting the timeout
+    exhaustMap(([, { gasPriceFactor }]) =>
+      defer(async () => provider.getGasPrice()).pipe(
+        map((price) => price.mul(Math.round(gasPriceFactor * 1e6)).div(1e6)),
+        map((price) => (price.gte(1e9) ? price : 1e9)), // min 1Gwei
```

Comment:
Why provide this minimum?

Reply:
I've seen ETH node tripping balls sometimes and giving very low gasPrices (like 1wei) on testnets, which even there made txs stuck for several hours; this just ensures a reasonable minimum which is compatible with most networks.

Reply:
Thanks for the explanation. I was a bit scared that it's to high for testnets, but there it also doesn't matter if spending a bit more.